_posts/2020-06-07-higher-order-hydras.markdown

```diff
@@ -0,0 +1,491 @@
+> Let \\\(D'\\\) be \\\(D\\\) but with the label changed to \\\(\star\\\).
+> Let \\\(E'\\\) be \\\(S(D')\\\) but without children whose label is \\\((\star:)\\\) and with the children of \\\(E\\\) whose labels are \\\((\star:)\\\) appended as children, and with the label of \\\(D\\\).
+> Let \\\(F = ((\star:):)\\\).
+> Do this \\\(N\\\) times, for some \\\(N\\\): "Replace \\\(F\\\) with \\\(E'\\\) where \\\(D\\\) is replaced by \\\(F\\\)."
+> Replace \\\(D\\\) with \\\(F\\\).
+> \\\(S(A)\\\) is the modified \\\(A\\\).
+>
+> **2.** If \\\(C > (\star:)\\\):
+> Go right in the tree, looking for the rightmost leaf.
+> If \\\(C\\\) has no children the search just ends there.
+>
+> **2.1.** If a node \\\(D\\\) with label \\\((\star:)\\\) is encountered, proceed as in rule **1**.
+>
+> **2.2** If we reach a leaf without ever encountering a node with label \\\((\star:)\\\):
+> Let \\\(D\\\) be this rightmost leaf.
+> Let \\\(E\\\) be the label of \\\(D\\\).
+> Traverse up the parents from \\\(D\\\) searching for a node \\\(F\\\) with \\\(G\\\) as the label of \\\(F\\\) satisfying \\\(G < E\\\), which is always guaranteed to exist, because the root always satisfies this.
+> Let \\\(H = S(E)\\\).
+> Let \\\(F'\\\) be \\\(F\\\) but with the label changed to \\\(H\\\).
+> Let \\\(I = ((\star:):)\\\).
+> Do this \\\(N\\\) times, for some \\\(N\\\): "Replace \\\(I\\\) with \\\(F'\\\) where \\\(D\\\) is replaced by \\\(I\\\)"
+> Replace \\\(D\\\) with \\\(I\\\).
+> \\\(S(A)\\\) is the modified \\\(A\\\).
+
+The rules are sensitive to the order of recursive evaluations if the value of \\\(N\\\) depends on which comes first, as it does in the derived fast-growing function.
+In the future I may need to re-order the instructions to better reflect the underlying ordinal notation, though for now this is okay.
+
+If you want to intuitively understand what each rule does:
+
+* **1.1** is the Kirby-Paris chop and regrow rule, which Buchholz hydras also have.
+* **1.2** is an adaptation of the Buchholz hydra successor stack rule. You can see the correspondence if you un-externalize the higher labelled children back into the label somehow.
+* **2.1** redirects to **1**. It's perhaps not the strongest variant of the rule that's compatible with the other rules, but it is very safe (in that it won't accidentally make the hydra ordinal much smaller), and produces the intended behaviour as far as I know.
+* **2.2** is yet another rule adapted from the Buchholz hydra successor stack rule, so perhaps you can think of it like a tier 2 Buchholz successor stack rule. It's also responsible for reducing the labels and expanding those nodes out. It used to be weaker, and expand more like the Veblen functions, but I thought of this variant and I'm pretty sure it still always terminates so this is what I'm publishing first.
```

Comment:
I really like this section! I find it effective in summarizing your rules for your outer reduction function.

Reply:
Thank you